### PR TITLE
I2c rename, small docs clean up

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

### Pull Request Details 📖

Just some bits I noticed reviewing our docs, the important one being the `I2C` => `I2c` change.

### CODE REVIEW

File: esp-hal/src/prelude.rs [hunk 1]
```diff
@@ -6,36 +6,46 @@
 //! things, particularly traits, which are used in almost every single Rust
 //! program.
 
-pub use fugit::{ExtU64 as _fugit_ExtU64, RateExtU32 as _fugit_RateExtU32};
-pub use nb;
+pub use prelude::*;
```
[the remark this answers is not in the document]
Comment (MabezDev) [reply esp-hal/src/prelude.rs:9]: image

Imo seeing what's actually in the prelude in the docs isn't that helpful, but I'm happy to revert if you feel differently.

[the remark this answers is not in the document]
Comment (bugadani) [reply esp-hal/src/prelude.rs:9]: Eh whatever, I'd probably refrain from reexporting any of the things that you hid anyway.

**Comment by MabezDev** [comment]
Skip changelog for fixing esp-wifi lint

**Review by bugadani — APPROVED** [review]

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks!

**Review by bjoernQ — APPROVED** [review]
LGTM

**Pull request merged**